esmvaltool/preprocessor/_multimodel.py

```diff
@@ -13,27 +31,162 @@
+        statistic_function = np.ma.mean
+    else:
+        raise NotImplementedError
+
+    # no plevs
+    if len(datas[0].shape) < 3:
+        statistic = statistic_function(datas, axis=0)
+        return statistic
+
+    # plevs
+    for j in range(statistic.shape[0]):
+        len_stat_j = sum(1 for _ in
+                         (_plev_fix(cdata, j)
+                          for cdata in datas
+                          if _plev_fix(cdata, j) is not None))
+        stat_all = np.ma.zeros((len_stat_j,
+                                statistic.shape[1],
+                                statistic.shape[2]))
+        for i, e_l in enumerate((
+                _plev_fix(cdata, j)
+                for cdata in datas
+                if _plev_fix(cdata, j) is not None)):
+            stat_all[i] = e_l
+
+        # check for nr models
+        if len_stat_j >= 2:
+            statistic[j] = statistic_function(stat_all, axis=0)
+        else:
+            mask = np.ones(statistic[j].shape, bool)
+            statistic[j] = np.ma.array(statistic[j],
+                                       mask=mask)
+
+    return statistic
+
+
+def _put_in_cube(template_cube,
+                 cube_data,
+                 stat_name,
+                 file_name,
+                 t_axis):
+    """Quick cube building and saving"""
+    # grab coordinates from any cube
+    times = template_cube.coord('time')
+    # or get the FULL time axis
+    if t_axis is not None:
+        times = iris.coords.DimCoord(
+            t_axis, standard_name='time',
+            units=template_cube.coord('time').units)
+    lats = template_cube.coord('latitude')
+    lons = template_cube.coord('longitude')
+
+    # no plevs
+    if len(template_cube.shape) == 3:
+        cspec = [(times, 0), (lats, 1), (lons, 2)]
+    # plevs
+    elif len(template_cube.shape) == 4:
+        plev = template_cube.coord('air_pressure')
+        cspec = [(times, 0), (plev, 1), (lats, 2), (lons, 3)]
+
+    # correct dspec if necessary
+    fixed_dspec = np.ma.fix_invalid(cube_data,
+                                    copy=False,
+                                    fill_value=1e+20)
+    # put in cube
+    stats_cube = iris.cube.Cube(fixed_dspec,
+                                dim_coords_and_dims=cspec,
+                                long_name=stat_name)
+    coord_names = [coord.name() for coord in template_cube.coords()]
+    if 'air_pressure' in coord_names:
+        if len(template_cube.shape) == 3:
+            stats_cube.add_aux_coord(template_cube.
+                                     coord('air_pressure'))
+    stats_cube.attributes['_filename'] = file_name
     return stats_cube
 
 
-def _sdat(srl_no):
-    """convert to a datatime point"""
-    new_date = dd(1950, 1, 1, 0) + td(srl_no)
+def _to_datetime(delta_t, unit_type):
+    """Convert to a datetime point"""
+    if unit_type == 'day since 1950-01-01 00:00:00.0000000':
+        new_date = dd(1950, 1, 1, 0) + td(delta_t)
```

Comment:
Use `datetime` instead of `dd` and `timedelta` instead of `td`, this is hard to read.